lib/graphiti/active_graph/extensions/grouping/params.rb

```diff
@@ -0,0 +1,56 @@
+module Graphiti::ActiveGraph::Extensions::Grouping
+  class Params
+    attr_reader :params, :grouping_criteria_list, :resource_class
+    def initialize(params, resource_class)
+      @params = params
+      @grouping_criteria_list = params.fetch(:group_by, nil)&.split(',') || []
+      @resource_class = resource_class
+    end
+
+    def single_grouping_criteria?
+      grouping_criteria_list.size < 2
+    end
+
+    def grouping_criteria_on_attribute?
+      grouping_criteria_list.any? { |criteria| ends_with_attribute?(resource_class.model, criteria) }
+    end
+
+    def empty?
+      grouping_criteria_list.empty?
+    end
+
+    def ends_with_attribute?(model, criteria)
+      return false if criteria.blank?
+
+      segments = criteria.split('.')
+      traverse_segments(model, segments)
+    end
+
+    private
+
+    def traverse_segments(model, segments)
```

Comment:
this is a bit less descriptive then how you usually write. Here the return type is Boolean but name suggest some side effects. How this should work? 
when :employee fallback to false line:40
when :employee.name to true line:34
when :opportunity.name to false line:37
?